cpp/src/arrow/compute/kernel.h

```diff
@@ -293,6 +298,10 @@ class ARROW_EXPORT OutputType {
   OutputType(Resolver resolver)  // NOLINT implicit construction
       : kind_(COMPUTED), resolver_(std::move(resolver)) {}
 
+  /// \brief For backward compatibility
+  OutputType(ResolverFuncPtr resolver)  // NOLINT implicit construction
+      : kind_(COMPUTED), resolver_(std::move(resolver)) {}
```

Comment:
Since the constructors are not marked explicit, many existing codes implicitly pass a function pointer as `OutputType`. To make them compile, I'll have to either add this constructor or change all occurences. I chose the easier path.

Reply:
Nevermind, some compilation fails due to ambiguous call. I guess I'll have to change all occurences.

Reply:
Actually making this a templated method is more elegant. Problem solved.